client/src/main/java/io/split/client/SplitClientImpl.java

```diff
@@ -123,111 +132,277 @@ public Map<String, String> getTreatments(String key, List<String> featureFlagNam
         return getTreatmentsBySetsWithConfigInternal(key, null, new ArrayList<>(Arrays.asList(flagSet)),
-                null, MethodEnum.TREATMENTS_WITH_CONFIG_BY_FLAG_SET);
+                null, new EvaluationOptions(null), MethodEnum.TREATMENTS_WITH_CONFIG_BY_FLAG_SET);
     }
 
     @Override
     public Map<String, SplitResult> getTreatmentsWithConfigByFlagSet(String key, String flagSet, Map<String, Object> attributes) {
         return getTreatmentsBySetsWithConfigInternal(key, null, new ArrayList<>(Arrays.asList(flagSet)),
-                attributes, MethodEnum.TREATMENTS_WITH_CONFIG_BY_FLAG_SET);
+                attributes, new EvaluationOptions(null), MethodEnum.TREATMENTS_WITH_CONFIG_BY_FLAG_SET);
     }
 
     @Override
     public Map<String, SplitResult> getTreatmentsWithConfigByFlagSet(Key key, String flagSet, Map<String, Object> attributes) {
         return getTreatmentsBySetsWithConfigInternal(key.matchingKey(), key.bucketingKey(), new ArrayList<>(Arrays.asList(flagSet)),
-                attributes, MethodEnum.TREATMENTS_WITH_CONFIG_BY_FLAG_SET);
+                attributes, new EvaluationOptions(null), MethodEnum.TREATMENTS_WITH_CONFIG_BY_FLAG_SET);
     }
 
     @Override
     public Map<String, SplitResult> getTreatmentsWithConfigByFlagSets(String key, List<String> flagSets) {
         return getTreatmentsBySetsWithConfigInternal(key, null, flagSets,
-                null, MethodEnum.TREATMENTS_WITH_CONFIG_BY_FLAG_SETS);
+                null, new EvaluationOptions(null), MethodEnum.TREATMENTS_WITH_CONFIG_BY_FLAG_SETS);
     }
 
     @Override
     public Map<String, SplitResult> getTreatmentsWithConfigByFlagSets(String key, List<String> flagSets, Map<String, Object> attributes) {
         return getTreatmentsBySetsWithConfigInternal(key, null, flagSets,
-                attributes, MethodEnum.TREATMENTS_WITH_CONFIG_BY_FLAG_SETS);
+                attributes, new EvaluationOptions(null), MethodEnum.TREATMENTS_WITH_CONFIG_BY_FLAG_SETS);
     }
 
     @Override
     public Map<String, SplitResult> getTreatmentsWithConfigByFlagSets(Key key, List<String> flagSets, Map<String, Object> attributes) {
         return getTreatmentsBySetsWithConfigInternal(key.matchingKey(), key.bucketingKey(), flagSets,
-                attributes, MethodEnum.TREATMENTS_WITH_CONFIG_BY_FLAG_SETS);
+                attributes, new EvaluationOptions(null), MethodEnum.TREATMENTS_WITH_CONFIG_BY_FLAG_SETS);
+    }
+
+    @Override
+    public String getTreatment(String key, String featureFlagName, EvaluationOptions evaluationOptions) {
+        return getTreatment(key, featureFlagName, Collections.<String, Object>emptyMap(), evaluationOptions);
+    }
+
+    @Override
+    public String getTreatment(String key, String featureFlagName, Map<String, Object> attributes, EvaluationOptions evaluationOptions) {
+        return getTreatmentWithConfigInternal(key, null, featureFlagName, attributes, evaluationOptions, MethodEnum.TREATMENT).treatment();
```

Comment:
why are you using `getTreatmentWithConfigInternal` here and `getTreatment` in line 246? what is the difference?

Reply:
The getTreatment in 246 does not have attributes param, it is calling getTreatment in 250 with attributes null.